mlir/dialects/func.py

```diff
@@ -23,7 +23,7 @@ class CallIndirectOperation(DialectOp):
 @dataclass
 class CallOperation(DialectOp):
     func: mast.SymbolRefId
-    type: mast.FunctionType
+    type: Optional[mast.FunctionType] = None
```

Comment:
should there not be an accompanying lark change for this? Also, why would calls not have type information? Is that a new addition to MLIR?

Reply:
This was probably a copy-paste problem with argtypes being a duplicate of type. Will clean up the unused argtypes in CallIndirectOperation too.